src/app/services/weather-api.service.ts

```diff
@@ -27,4 +25,23 @@ export class WeatherAPIService {
   private requestWeather () {
     return this.http.get<Weather>(API_URL)
   }
+
+  private countByDirectionVal (arr: Array<Data> | undefined, val: string) {
+    let i = 0
+    arr?.map((v) => v.wind10m.direction === val ? i++ : i)
+    return i
+  }
+
+  groupWind () {
+    const result = new Map<string, number>()
+    this.getWeather().forEach(v => {
+      v.dataseries?.map((el) =>
+        result.set(el.wind10m.direction, (result.get(el.wind10m.direction) || 0) + el.wind10m.speed)
+      )
+      result.forEach((value: number, key: string) => {
+        result.set(key, value / this.countByDirectionVal(v.dataseries, key))
+      })
+    })
```

Comment:
if to analyze this part:
- you need one of the property from v: Weather - it's just dataseries, and based on it you're creating some aggregation in Map,


Let's simplify the goal:
- we need to return still observable, but it will be normalized observable that will return your map. So the end goal for refactoring this PR will be next:
``` groupWind (): Observable<Map<string, number>> {
        return this.getWeather().pipe()
        }
```
and knowing this, you can reuse this.getWeather() observable with some normalization pipes around.

  
one of the solution might be simple, you just need map operator(https://www.learnrxjs.io/learn-rxjs/operators/transformation/map) to access the v: Weather and create result map inside and return it:

``` groupWind (): Observable<Map<string, number>> {
        return this.getWeather().pipe(map( ( v: Weather): Map<string, number> => {
        
       // you're free to create result = new Map here and return it from there
        }))
        }
```